src/main/java/Controller/newPlaylistController.java

```diff
@@ -0,0 +1,46 @@
+package Controller;
+import Entities.RegularUser;
+import Entities.Song;
+import InputBoundary.newPlaylistInputBoundary;
+import InputData.newPlaylistInputData;
+/**
+ * Controller-layer implementation for create playlist use case
+ */
+public class newPlaylistController {
+    private final newPlaylistInputBoundary newPlaylistInputBoundary;
+
+    /**
+     *
+     * @param createPlaylistInteractor interactor object to be passed into controller for use case
```

Comment:
This is assuming that we will be passing an interactor into the controller, which is not the way we've been working in previous controllers (thus far we have been constructing them inside the controller). The other option is to construct the interactor in the view model, which it appears is what you are doing. While this is valid, and will work (and might even be the better choice tbh, now that I think about it), perhaps we should talk about this as a group and decide on one method or the other for consistency across our codebase.

Reply:
For now, no changes need to be made - it works and is a 'correct' implementation by clean architecture standards. I think I'll even lobby for changing our other controller implementations to follow this one - it's less clutter since we don't have to ever give the controller the presenter. Thanks for making me realize this lol